src/Workspaces/CSharpTest/Formatting/FormattingTests_FunctionPointers.cs

```diff
@@ -31,19 +30,90 @@ unsafe class C
         }
 
         [Fact]
-        public async Task FormatFunctionPointerWithCallingConvention()
+        public async Task FormatFunctionPointerWithManagedCallingConvention()
         {
-            // TODO(https://github.com/dotnet/roslyn/issues/44312): add a space after the "int"s in the baseline and make this test still pass
             var content = @"
 unsafe class C
 {
-    delegate *cdecl < int,  int> functionPointer;
+    delegate *managed < int ,  int > functionPointer;
 }";
 
             var expected = @"
 unsafe class C
 {
-    delegate* cdecl<int, int> functionPointer;
+    delegate* managed<int, int> functionPointer;
+}";
+
+            await AssertFormatAsync(expected, content);
+        }
+
+        [Fact]
+        public async Task FormatFunctionPointerWithUnmanagedCallingConvention()
+        {
+            var content = @"
+unsafe class C
+{
+    delegate *unmanaged < int ,  int > functionPointer;
+}";
+
+            var expected = @"
+unsafe class C
+{
+    delegate* unmanaged<int, int> functionPointer;
+}";
+
+            await AssertFormatAsync(expected, content);
+        }
+
+        [Fact]
+        public async Task FormatFunctionPointerWithUnmanagedCallingConventionAndSpecifiers()
+        {
+            var content = @"
+unsafe class C
+{
+    delegate *unmanaged [ Cdecl ,  Thiscall ] < int ,  int > functionPointer;
+}";
+
+            var expected = @"
+unsafe class C
+{
+    delegate* unmanaged[Cdecl, Thiscall]<int, int> functionPointer;
+}";
+
+            await AssertFormatAsync(expected, content);
+        }
+
+        [Fact(Skip = "PROTOTYPE(func-ptr): Even though the rule is returning put a space, it's getting ignored, resulting in delegate*invalid <")]
```

Comment:
I'm assuming that this is because `invalid` is parsed as trivia on a missing `managed` or `unmanaged` token? Those two are the only valid token types for the spot.

Reply:
The "invalid" token isn't being parsed into the calling convention position? It looks based on the formatting change that's what you intended, right?

Reply:
It is parsed into the calling convention position, it's parsed as trivia on a missing `managed` token.

---
In reply to: [451061448](https://github.com/dotnet/roslyn/pull/45684#discussion_r451061448) [](ancestors = 451061448)

Reply:
So I guess if the formatter's rule is we don't touch things around missing stuff, then maybe the behavior you're seeing is correct. Once the user has typed _correct_ code, it'll format correctly; if the user put their code into a broken state while they're doing other editing, not touching is fine for that small window.